DSE 6.8.22 Release Notes (#56)
datastax/release-notes: DSE_6.8_Release_Notes.md

## 6.8.22 DSE Cassandra
* Limit heap pressure during mutation repair for tables with materialized views by throttling number of concurrent batches (default 10). Number of batches can be controlled by new system property {{cassandra.repair.mutation_repair_max_concurrent_batches}}. Setting to 0 (zero) disables throttling and reverts behavior before this change. (DSP-22344)

## 6.8.22 DSE Graph
* Replaced log4j with reload4j in TinkerPop and bumped the version of TinkerPop. (DSP-22326)

## 6.8.22 DSE NodeSync
* Command line tools will automatically use a heap size of 512M on Azul Zing if MAX_HEAP_SIZE is not specified. (DSP-22313)

## 6.8.22 DSE Security
* Upgrades Bouncy Castle to the latest 1.70 version. (DSP-22352)

## 6.8.22 DSE Miscellaneous
* Ported fix from DSP-22315: Option to disable call to NativeLibrary.trySkipCache. (DSP-22343)

## 6.8.22 DSE CVE

> * Upgrade azure-storage-blob from 12.4.0 to 12.15.0. Removes CVE-2020-5403. (DSP-22377, [CVE-2020-5403](https://nvd.nist.gov/vuln/detail/CVE-2020-5403))

Author: I have rephrased RNs in all CVE tickets

DSE 6.8.x is compatible with Apache Cassandra&trade; 3.11 and adds additional production-certified changes, if any. Components that are indicated with an asterisk (&ast;) (if any) are known to be updated since the prior patch version.

# Release notes for 6.8.22
11 April 2022

## 6.8.22 DSE Platform
* Introduces DSE support for Java 11 for core Cassandra workloads. Please note that DSE does not currently provide support for advanced workloads (Search, Spark and Graph) for Java 11.

## Components versions for DSE 6.8.22
 * Apache Solr™ 6.0.1.4.2887
 * Apache Spark™ 2.4.0.18
 * Apache TinkerPop™ 3.4.5-20220405-a52bbe2c&ast;
 * Apache Tomcat® 8.5.75&ast;
 * DSE Java Driver 1.10.0-dse+20210424
 * Netty 4.1.34.3.dse&ast;
 * Spark JobServer 0.8.0.50

## 6.8.22 DSE Core
* Adds a startup check that prevents starting DSE with advanced workloads on Java 11. (DSP-22358)
* Upgrade netty to 4.1.34 from 4.1.25. (DSP-22363)
* Removes {{disk_cal.py}} and {{compaction-metrics.ipynb}} Python 2 tools. (DSP-22382)
* Use azure sdk client libraries from bill of materials 1.2.0 for Azure Blob Storage backup and restore. Use okhttp based azure code http client instead of the netty based one. (DSP-22401)

> * Node bootstrapping and rebuild operation require to calculate a map of nodes used as sources while streaming data. There are two algorithms in use, simplified that picks the first closest candidate and more advanced that uses Ford-Fulkerson graph algorithm. For `Everywhere` replication strategy we don't have to start a time-consuming graph algorithm as each node contains the data we want to stream. The fix ensures a simpler (and faster) algorithm is used for keyspaces that use Everywhere replication. (DSP-22339)

Author: ```suggestion
* Reduce node bootstrapping and rebuild operation time by improving algorithms that calculate streaming candidates' map. (DSP-22339)
```
